books/RayTracingInOneWeekend.html

```diff
@@ -502,54 +502,103 @@
+    image_height = (image_height < 1) ? 1 : image_height;
+
+    // Calculate the viewport height. Since it's real valued, we don't check for values less then 1.
+    auto viewport_height = 2.0;
+    auto viewport_width = viewport_height * (static_cast<double>(image_width)/image_height);
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [image-setup]: Rendered image setup]
+
+(If you're wondering why we don't just use `aspect_ratio` when computing `viewport_width`, it's
+because the _actual_ aspect ratio is the integer-based ratio of image width over image height, which
+will yield zero for values less than one.)
 
-I’ll put the “eye” (or camera center if you think of a camera) at $(0,0,0)$.
+Next we will define the camera center: a point in 3D space from which all scene rays will originate
+  (this is also commonly referred to as the _eye point_).
+The vector from the camera center to the viewport center will be orthogonal to the viewport.
+We'll initially set the distance between the viewport and the camera center point to be one unit.
+This distance is often referred to as the _focal length_.
+
+I’ll put the camera center at $(0,0,0)$.
 I will have the y-axis go up, and the x-axis to the right.
-In order to respect the convention of a right handed coordinate system, into the screen is the
-negative z-axis.
+Since we use a right-handed coordinate system, the vector into the screen is the negative z-axis.
+Note that I do not make the ray direction a unit length vector because I think not doing that makes
+  for simpler and slightly faster code.
+
+  ![Figure [camera-geom]: Camera geometry](../images/fig-1.03-cam-geom.jpg)
 
 Now the inevitable tricky part.
 While our 3D space has the conventions above, this conflicts with the standard _image_ coordinates,
-where the Y axis is inverted.
-In image coordinates, Y=0 denotes the top of the image, and Y=1 denotes the bottom.
-
-To help keep things straight, we'll compute two normalized (that is, in the range [0,1]) image
-coordinates: $s$ (horizontal) and $t$ (vertical).
-$s=0$ is image left, $s=1$ is image right, $t=0$ is image top, and $t=1$ is image bottom.
-That means that we get our vertical axis flip at the right point by computing $t = 1 - y$, where $y$
-is the vertical component of our 3D coordinate system, and $t$ is the flipped vertical component of
-our image coordinate system.
-
-I will traverse the screen from the upper left hand corner, and use two offset vectors along the
-screen sides to move the ray endpoint across the screen.
-Note that I do not make the ray direction a unit length vector because I think not doing that makes
-for simpler and slightly faster code.
+  where the Y axis is inverted (increasing going down).
+
+As we scan our image, we will start at the upper left pixel (pixel 0,0), scan left-to-right across
+  the row, and then scan the next row, top-to-bottom (reading order for most Latin languages).
```

Comment:
As we can our image, we will start at the upper-left pixel (pixel \math(0,0)) scan left-to-right across the row, and then scan one row after another from the top to the bottom

I would remove (reading order for most Latin languages)
I'm not sure it's helpful and it's off topic

Reply:
Changed to:
> As we scan our image, we will start at the upper left pixel (pixel $0,0$), scan left-to-right across
>   each row, and then scan row-by-row, top-to-bottom.